src/taipy/gui/gui.py

```diff
@@ -408,10 +408,7 @@ def _update_var(
         propagate=True,
         holder: t.Optional[_TaipyBase] = None,
         on_change: t.Optional[str] = None,
-        from_map_dict: bool = False,
     ) -> None:
-        if from_map_dict:
-            var_name = _variable_encode(var_name, self._get_locals_context())
         if holder:
             var_name = holder.get_name()
         hash_expr = self.__evaluator.get_hash_from_expr(var_name)
```

Comment:
I guess the calculation of the right variable name taht is done in __call_on_change should be done somewhere here ;-)

Reply:
It is not neccessary anymore I think, I dont remember why I add that but that was when everything was a mess :D